pkg/apis/policy/types.go

```diff
@@ -444,6 +449,21 @@ const (
 	SupplementalGroupsStrategyRunAsAny SupplementalGroupsStrategyType = "RunAsAny"
 )
 
+// RuntimeClassStrategyOptions define the strategy that will dictate the allowable RuntimeClasses
+// for a pod.
+type RuntimeClassStrategyOptions struct {
+	// AllowedRuntimeClassNames is a whitelist of RuntimeClass names that may be specified on a pod.
+	// An empty string value allows the RuntimeClassName to be left unset.
+	// An empty list allows all RuntimeClasses.
```

Comment:
If runtimeClass is present in a PSP, I think being more explicit is helpful.

include `*` in allowedRuntimeClassNames to allow all. For consistency with `volumes`, `*` can be alongside explicit values

To default and limit:
```
runtimeClass: {
  "allowedRuntimeClassNames": ["foo","bar"],
  "defaultRuntimeClassName: "foo"
}
```

To default and not limit:
```
runtimeClass: {
  "allowedRuntimeClassNames": ["*"],
  "defaultRuntimeClassName: "foo"
}
```

To limit and not default:
```
runtimeClass: {
  "allowedRuntimeClassNames": ["foo","bar"]
}
```

Reply:
runtimeClass would be optional/omitempty
allowedRuntimeClassNames is required
defaultRuntimeClassName is optional/omitempty